test/common/http/http1/codec_impl_test.cc

```diff
@@ -81,6 +81,43 @@ struct HTTPStringTestCase {
   const absl::optional<absl::string_view> http_parser_expected_error_;
 };
 
+class MockRequestDecoderWithUhv : public Http::MockRequestDecoder {
```

Comment:
This reads as a header decoder using UHV but as it compiles and presumably works without UHV I assume it's more of a wrapper which handles UHV more consistently?  It'd be helpful to have better naming and maybe a class comment.  Also will this be useful outside of this test and should be base class functionality or really only for codec corner cases?
/wait

Reply:
Clarified comments.